src/helpers.cpp

```diff
@@ -151,4 +151,182 @@ Vec3 Triangulate2(const SE3 &g12, const Vec2 &xc1, const Vec2 &xc2) {
   return x;
 }
 
+
+Vec3 Triangulate3(const SE3 &g12, const Vec2 &xc1, const Vec2 &xc2) {
+
+  // Initalize the Rotation and Translation Matricies
+  Vec3 t12{g12.T()};
+  Mat3 R12{g12.R()};
+  Mat3 R21{R12.transpose()};
+  Vec3 t21{-1 * R12.transpose() * t12};
+
+  // Create homogeneous coordinates
+  Vec3 f0{xc1(0), xc1(1), 1.0};
```

Comment:
The paper uses `0` and `1`, while we use `1` and `2`. Having two conventions within one function is confusing. I would suggest making this entire function use the convention of the paper (`0` and `1`), and in `Feature::Triangulate`, we just pass in `xc1` and `xc2`.

Reply:
Fixed!

Reply:
Fixed!